sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -436,12 +438,22 @@ public String getEntityPath() {
                 String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "renewMessageLock")));
         } else if (Objects.isNull(lockToken)) {
             return monoError(logger, new NullPointerException("'receivedMessage' cannot be null."));
+        } else if (CoreUtils.isNullOrEmpty(lockToken.getLockToken())) {
+            return monoError(logger, new NullPointerException("'receivedMessage.lockToken' cannot be null."));
+        }
+
+        UUID lockTokenUUID;
+        try {
+            lockTokenUUID = UUID.fromString(lockToken.getLockToken());
+        } catch (IllegalArgumentException ex) {
+            return monoError(logger, new IllegalStateException(
```

Comment:
shouldn't we propagate the same error back? It is an illegalargument exception because it couldn't be converted to a UUID. It's not dependent on the state of this receiver.